Add Notification dropdown by status

SferaDev opened the pull request:
### :pushpin: References

* **Issue:** Closes #96 

### :memo: Implementation

- Notifications Page: Add filter by status in a dropdown

### :art: Screenshots


### :fire: Is there anything the reviewer should know to test it?


### :bookmark_tabs: Others

tokland commented on i18n/en.pot:
```diff
@@ -146,6 +146,21 @@ msgstr ""
+msgstr ""
+
+msgid "Synchronization Status"
+msgstr ""
```
[Minor] We show lowercase here but uppercase in the table:

![image](https://user-images.githubusercontent.com/24643/58706195-9abc6a00-83b1-11e9-9c82-2cfbd9f908c3.png)

tokland replied:
Also, the dropdown is very close to the title. This didn't happen before because we had the search bar. Not sure if this should/can be fixed.

SferaDev replied:
We noticed with Adrian too the spacing, will add some margin to top.

tokland commented on src/components/notifications-table/NotificationsTable.jsx:
```diff
@@ -141,8 +145,50 @@ class NotificationsTable extends React.Component {
+                name: i18n.t("Done"),
+            },
+        ];
+        this.setState({ statusFilter: { value, items } });
```
All info is know, can we move it to the initial code?

SferaDev replied:
Yes! I mirrored the logic of the filters in BaseSyncConfigurator where the items are loaded async 😑

SferaDev commented:
Fixed comments from pull request:

- Added the missing d2, it's true it's solved on another PR but doesn't hurt to merge it now.

- The ObjectsTable now shows errors properly "Done" instead of "DONE".

- Added some spacing to top.

- Moved the dropdown value directly to state and items to internal prop

SferaDev commented:
@adrianq Yesterday you complained that we were getting "FAILURE" when the sync had "WARNINGS", in this PR we also fix that by only setting the "FAILURE" flag with straight errors (DHIS2 errors or network errors).
